spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientAutoConfiguration.java

```diff
@@ -48,8 +48,8 @@
 
 	@Bean
 	@ConditionalOnMissingBean
-	public DefaultIsServicePortSecureResolver isServicePortSecureResolver(KubernetesDiscoveryProperties properties) {
-		return new DefaultIsServicePortSecureResolver(properties);
+	public ServicePortSecureResolver isServicePortSecureResolver(KubernetesDiscoveryProperties properties) {
```

Comment:
I dont see why we need this defined as a bean, as since the class is not public no one could provide their own implementation anyways

Reply:
indeed. changed.